app/lib/efile/nj/nj1040_calculator.rb

```diff
@@ -358,6 +359,12 @@ def calculate_line_54
         [sum, 0].max
       end
 
+      def calculate_line_55
+        return nil if @intake.state_file_w2s.empty?
+
+        @intake.state_file_w2s.inject(0) { |sum, w2| sum + w2.state_income_tax_amount }.round
```

Comment:
[Pebble]: This might be a tax feature I am not aware of but is using `.round` here appropriate? That seems to remove any decimal value. Marking this a pebble because it could have a pretty large effect if not intentional

[Dust]: Also, curious what the `inject(0)` is doing here and if there is a more explicit way to pull out the values needed from the `state_file_w2s` object. (I don't know ruby super well so I could also just be reading this wrong 🙏 )

Reply:
good callout on the `round`.. we decided as a team to always round since that is what the dept. of taxation ultimately does anyway, so we round and then on the review page inform the user that we rounded.

and `inject()` is the same as `reduce()` in ruby, so if you're used to reduce then all is well. (i also had to look up since i'm used to the `reduce()` syntax)